addons/point_of_sale/tests/test_frontend.py

```diff
@@ -1855,6 +1855,102 @@ def test_product_create_update_from_frontend(self):
+            ptav.attribute_id == attributes[0] and ptav.product_attribute_value_id == attributes[0].value_ids[0]
+            for ptav in v.product_template_attribute_value_ids
+        ))[0]
+        variant2 = variants.filtered(lambda v: any(
+            ptav.attribute_id == attributes[0] and ptav.product_attribute_value_id == attributes[0].value_ids[1]
+            for ptav in v.product_template_attribute_value_ids
+        ))[0]
+
+        # Create a POS order using both variants to ensure they can't be deleted
+        self.main_pos_config.open_ui()
+        self.env['pos.order'].create({
+            'partner_id': self.env['res.partner'].create({'name': 'Test Partner'}).id,
+            'session_id': self.main_pos_config.current_session_id.id,
+            'amount_tax': 0,
+            'amount_total': 1,
+            'amount_paid': 1,
+            "amount_return": 0,
+            'lines': [
+                (0, 0, {
+                    'product_id': variant1.id,
+                    'price_unit': variant1.lst_price,
+                    'qty': 1,
+                    'discount': 0.0,
+                    'tax_ids': False,
+                    'price_subtotal': variant1.lst_price,
+                    'price_subtotal_incl': variant1.lst_price,
+                }),
+                (0, 0, {
+                    'product_id': variant2.id,
+                    'price_unit': variant2.lst_price,
+                    'qty': 1,
+                    'discount': 0.0,
+                    'tax_ids': False,
+                    'price_subtotal': variant1.lst_price,
+                    'price_subtotal_incl': variant1.lst_price,
+                })
+            ]
+        })
+
+        # Store variant ids for later checking
+        variant_ids = [variant1.id, variant2.id]
+
+        # Remove first attribute - this should archive variants that differ only by this attribute
+        template.attribute_line_ids.filtered(lambda l: l.attribute_id == attributes[0]).unlink()
+
+        # Archive a specific variant to test UI warning in the product configurator
+        # This variant has Value 2-B and Value 3-B (see tour steps)
+        variant_to_archive = template.product_variant_ids.filtered(
+            lambda v: any(ptav.product_attribute_value_id == attributes[1].value_ids[1] for ptav in v.product_template_attribute_value_ids)
+                      and any(ptav.product_attribute_value_id == attributes[2].value_ids[1] for ptav in v.product_template_attribute_value_ids)
+        )
+        variant_to_archive.write({'active': False})
+
+        # Verify variants from order still exist but are archived
+        archived_variants = self.env['product.product'].with_context(active_test=False).browse(variant_ids)
+        self.assertTrue(archived_variants.exists(), "Variants used in sale order should still exist")
+        self.assertFalse(all(archived_variants.mapped('active')), "Variants used in sale order should be archived")
```

Comment:
Note that `all([False, True]) = False`. I think the following check is what you want to do. 

```suggestion
        self.assertTrue(all(not active for active in archived_variants.mapped('active')), "Variants used in sale order should be archived")
```

Reply:
Nice catch, thanks!